code/game/objects/items/grenades/grenade.dm

```diff
@@ -189,6 +189,9 @@
 		log_game("A projectile ([hitby]) detonated a grenade held by [key_name(owner)] at [COORD(source_turf)]")
 		message_admins("A projectile ([hitby]) detonated a grenade held by [key_name_admin(owner)] at [ADMIN_COORDJMP(source_turf)]")
 		detonate()
+		
+		if(src && !QDELETED(src)) // some grenades don't detonate but we want them destroyed
```

Comment:
```suggestion
		if(!QDELETED(src)) // some grenades don't detonate but we want them destroyed
```
`QDELETED(src)` already implies `!src`, so you don't need to check for `src`

Reply:
We are checking two things here:

`src` is to check if the grenade is a dud and didn't detonate and `!QDELETED(src)` to make sure it wasn't garbage collected when it detonated.  Not sure if `!QDELETED` does both of these already?  `QDELETED` means an object is in the queue to get deleted if I'm not mistaken?

Reply:
Look at the macro for `QDELTED` -
`QDELTED(X) = !X || QDELING(X)`
`!QDELETED` says "do we exist or have we not qdeleted"